core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala

```diff
@@ -291,6 +297,16 @@ private[spark] trait SparkListenerInterface {
   def onBlockUpdated(blockUpdated: SparkListenerBlockUpdated): Unit
 
   /**
+   * Called when a speculative task is submitted
+   */
+  def onSpeculativeTaskSubmitted(speculativeTask: SparkListenerSpeculativeTaskSubmitted): Unit
+
+  /**
+   * Called when an extra executor is needed
+   */
+  def onExtraExecutorNeeded(): Unit
```

Comment:
@cloud-fan after thoughts, yes, I think we can get rid of extraExecutorNeeded event and handle it in ExecutorAllocationManager.scala.